gamemodes/terrortown/gamemode/server/sv_player.lua

```diff
@@ -1446,6 +1453,49 @@ function GM:PlayerTakeDamage(ent, infl, att, amount, dmginfo)
 	end
 end
 
+function HandlePlayerArmorSystem(ent, infl, att, amount, dmginfo)
+	-- armor does not help when the head receives damage
+	--if ent:LastHitGroup() == HITGROUP_HEAD then return end
+	-- DISABLED for now since the hook gets called twice, once with 0 damage, but with a hitgroup and once
+	-- with damage but no hitgroup. I have to dig deeper into this
+
+	-- some entities cause this hook to be triggered, ignore 0 damage events
+	if dmginfo:GetDamage() == 0 then return end
+
+	-- fallback for players who prefer the vanilla armor
+	if GetConVar("ttt_armor_classic"):GetBool() and ent:Armor() > 0 then
+		dmginfo:ScaleDamage(0.7)
+		
+		return
+	end
+
+	-- handle different damage type factors
+
+	if not dmginfo:IsDamageType(DMG_BULLET) and not dmginfo:IsDamageType(DMG_CLUB)
+		and not dmginfo:IsDamageType(DMG_BURN) and not dmginfo:IsDamageType(DMG_BLAST)
+	then return end
+
+	-- calculate damage
+	local damage = dmginfo:GetDamage()
+	local armor = ent:Armor()
+
+	-- normal damage handling when no armor is available
+	if armor == 0 then return end
+
+	local cv_armor_factor = GetConVar("ttt_armor_damage_block_pct"):GetFloat()
+	local cv_body_factor = GetConVar("ttt_armor_damage_health_pct"):GetFloat()
+	
+	if ent:ArmorIsReinforced() then
+		cv_body_factor = cv_body_factor - 0.15
+	end
+
+	armor = armor - cv_armor_factor * damage
```

Comment:
This would add armor for negative damage values.

Reply:
I have not thought about negative damage, thanks!